### PR TITLE
Fix typos in comments, add phpdocs, replace NBSP by space, simplify phpunit

### PR DESCRIPTION
**Pull request by kinow**
Happy to drop some commits if necessary :+1: Just syncing my local fork branch and taking a quick look at the current code base.

### CODE REVIEW

File: controller/Controller.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function setLanguageProperties($lang)
      * As a side effect, set the HTTP Vary header if a choice was made based on
      * the Accept header.
      * @param array $choices possible MIME types as strings
-     * @param string $accept HTTP Accept header value
+     * @param string $accept HTTP Accept header value
```
Comment (kinow) [inline controller/Controller.php:72]: Hmm, doesn't show on the GitHub UI. But we have a NBSP (unicode U+00A0), not a space here (U+0020).

File: model/Concept.php [hunk 2]
```diff
@@ -436,14 +436,14 @@ public function getMappingProperties(array $whitelist = null)
                             }
 
                             if ($response) {
-                                $ret[$prop]->addValue(new ConceptMappingPropertyValue($this->model, $this->vocab, $response, $this->resource, $prop), $this->clang);
```
Comment (kinow) [inline model/Concept.php:439]: Funny, these `clang` parameters are not in the `addValue` signature.

File: model/resolver/WDQSResource.php [hunk 3]
```diff
@@ -6,7 +6,6 @@ class WDQSResource extends RemoteResource
     const WDQS_ENDPOINT = "https://query.wikidata.org/sparql";
 
     public function resolve(int $timeout) : ?EasyRdf\Resource {
-        $client = new EasyRdf\Sparql\Client(self::WDQS_ENDPOINT);
```
Comment (kinow) [inline model/resolver/WDQSResource.php:9]: Re-declared in the code below

File: model/Concept.php (hunk 2 above)
Comment (kouralex) [reply model/Concept.php:439]: Yes, seems like the signature was changed some five years ago: https://github.com/NatLibFi/Skosmos/commit/2ba0a48174df87965e7785c8196ce2b039a85050#

Good catch!

File: controller/WebController.php [hunk 4]
```diff
@@ -588,10 +587,11 @@ public function getChangeList($request, $prop, $offset=0, $limit=200)
     }
 
     /**
-    * Formats the list of concepts as labels arranged by modification month
-    * @param Array $changeList
-    * @param string $lang the language for displaying dates in the change list
-    */
+     * Formats the list of concepts as labels arranged by modification month
+     * @param Array $changeList
```
Comment (kouralex) [inline controller/WebController.php:591]: It seems like there are PHP types stated in both uppercase and lowercase (starting with) in PHPDocs all over the code base (array, for instance, here).

Well, I guess they do not matter. 😅

File: model/resolver/WDQSResource.php (hunk 3 above)
Comment (kouralex) [reply model/resolver/WDQSResource.php:9]: Good catch!

Now that I look at it, it is a bit questionable why lines https://github.com/NatLibFi/Skosmos/blob/4774577479d618932ba68c44c2e59ff1af050c9d/model/resolver/WDQSResource.php#L10-L44 are in the `try` block, though. I would expect them to succeed in any case.

File: controller/WebController.php (hunk 4 above)
Comment (kinow) [reply controller/WebController.php:591]: I noticed that too. I **think** in this case I opened another code in this function, that returned `Array`, and then I just typed `Array` too :smile: but normally I use `string`, `boolean`, `array`…

File: model/resolver/WDQSResource.php (hunk 3 above)
Comment (kinow) [reply model/resolver/WDQSResource.php:9]: Hmm, good point. We can move it out maybe in another PR? One thing that really bothers me in the code is the use of `return $this->showError…` functions, while these `showError`-etc functions actually do not return anything; but instead set HTTP `header()` values, print to the stdout with `echo()` and need the stream to be closed (with `return` or `exit`) somewhere.

e.g.

- https://github.com/NatLibFi/Skosmos/blob/c84dc970cea2954c76c98254c20f72d6da33756a/controller/RestController.php#L49 returns the value of https://github.com/NatLibFi/Skosmos/blob/c84dc970cea2954c76c98254c20f72d6da33756a/controller/Controller.php#L179-L184 (which doesn't actually return anything).

File: tests/GenericSparqlTest.php [hunk 5]
```diff
@@ -1051,7 +1051,7 @@ public function testListConceptGroups()
     $voc = $this->model->getVocabulary('groups');
     $graph = $voc->getGraph();
     $sparql = new GenericSparql('http://localhost:13030/skosmos-test/sparql', $graph, $this->model);
-    $actual = $sparql->ListConceptGroups('http://www.w3.org/2004/02/skos/core#Collection', 'en', false);
+    $actual = $sparql->ListConceptGroups('http://www.w3.org/2004/02/skos/core#Collection', 'en');
```
Comment (kouralex) [inline tests/GenericSparqlTest.php:1054]: Good fix!

Oh my, PHP truly is case-insensitive. Hurts my eyes! 😄

File: model/resolver/WDQSResource.php (hunk 3 above)
Comment (kouralex) [reply model/resolver/WDQSResource.php:9]: Yes, best to leave it to another PR, this one is rock solid as it is! 😃 

Heh, now that you pointed at it, it surely looks ugly. I also remember having some trouble with those things whilst trying to silence the extra output PHPUnit was printing whilst executing tests.

+1 for another PR.

File: controller/WebController.php (hunk 4 above)
Comment (kouralex) [reply controller/WebController.php:591]: I guess it is OK at this point.

We should anyways add type hints/declarations directly into function arguments and return values as requested in #727 but that can be done en masse, simultaneously double checking these PHPDocs.

**Review by kouralex — APPROVED** [review]
Very solid improvements in this PR, thanks @kinow !

I am more than happy to merge this as-is.

**Comment by kinow** [comment]
Thanks for the review and helpful comments @kouralex !

**Comment by kouralex** [comment]
I merged this now that I saw your last replies. 🎉

**Pull request merged**